core/src/main/kotlin/configuration.kt

```diff
@@ -86,6 +86,20 @@ data class DokkaSourceSetID(
 
 fun DokkaConfigurationImpl(json: String): DokkaConfigurationImpl = parseJson(json)
 
+/**
+ * Global options are applied to all packages and modules and overwrite package configuration.
+ *
+ * These are handy if we have multiple sourcesets sharing the same global options as it reduces the size of the boilerplate.
+ * Otherwise, the user would be enforced to repeat all these options per each sourceset.
+ */
+data class GlobalDokkaConfiguration(
+    val perPackageOptions: List<PackageOptionsImpl>?,
+    val externalDocumentationLinks: List<ExternalDocumentationLinkImpl>?,
+    val sourceLinks: List<SourceLinkDefinitionImpl>?
+)
+
+fun GlobalDokkaConfiguration(json: String): GlobalDokkaConfiguration = parseJson(json)
+
```

Comment:
@IgnatBeresnev In order to not copy-paset whole json utilities, I had to move these to `core` since I cannot access `parseJson` from `core` being in `cli` module as it is `internal`. On the other hand, this `GlobalDokkaConfiguration` is not so `core` feature, rather `cli` specific, so it should not be there. Do you have any idea how we should deal with this problem? The third solution could be "opening" `parseJson` function in `core` module by removing `internal` keyword.

Reply:
To me, `GlobalDokkaConfiguration` seems to be a `json` feature rather than a `cli` one (although the only place we use json configuration is indeed in the cli), and since we already provide the ability to create dokka configuration from json up above, conceptually I'm fine with having this here

The only downside is that I, as a plugin developer, am unable to apply `GlobalDokkaConfiguration` to `DokkaConfigurationImpl` since this logic resides in `initializeConfiguration` (which is in CLI module):
```kotlin
        val globals = GlobalDokkaConfiguration(jsonContent)
        ...
        dokkaConfigurationImpl.apply {
            sourceSets.forEach {
                ...
            }
            ...
        }
```
What do you think about moving this to an extension function in core, say `fun DokkaConfiguration.apply(globalConfiguration: GlobalDokkaConfiguration)`?

If I'm writing a plugin, I then could use json configs like
```
val configuration = DokkaConfigurationImpl(mainJson)
val globalConfiguration = GlobalDokkaConfiguration(globalJson)

configuration.apply(globalConfiguration)
```

Reply:
I don't know if I got you clear, but do you propose to have two separate jsons?

Reply:
Not necessarily, could be one, could be separate, could be no json at all - that's up to the plugin writer.

My main proposed point was to add the ability to apply `GlobalDokkaConfiguration` to `DokkaConfiguration` (extract this code from `initializeConfiguration` in cli), then having this class in `core` is fine by me, since it can be reused by anyone

Reply:
Oh, ok, I got you wrong the first time I read your comment. Now it makes sense, will do fixes today.

Reply:
Nice, thanks!